src/test/java/com/exasol/swqa/maven/SummarizerMojoIT.java

```diff
@@ -0,0 +1,134 @@
+    private static final Path PLUGIN_JAR = Path.of("target", PLUGIN_JAR_FILE_NAME).toAbsolutePath();
+    // We need the flattened version of the POM file in order to include the generated POM.
+    private static final Path PLUGIN_POM = Path.of(".flattened-pom.xml").toAbsolutePath();
+    private static final Path BASE_TEST_DIR = Paths.get("src/test/resources").toAbsolutePath();
+
+    private static MavenIntegrationTestEnvironment testEnvironment;
+
+    @BeforeAll
+    // [itest -> dsn~maven-plugin~1]
+    static void beforeAll() {
+        testEnvironment = new MavenIntegrationTestEnvironment();
+        testEnvironment.installPlugin(PLUGIN_JAR.toFile(), PLUGIN_POM.toFile());
+    }
+
+    @CsvSource({ "10, 0, 0.0", "20, 80, 80.0", "20, 40, 66.7", "0, 100, 100.0", })
+    @ParameterizedTest
+    // [itest -> dsn~extracting-code-coverage-from-ja-co-co-report~1]
+    // [itest -> dsn~metric-output-file~1]
+    // [itest -> dsn~writing-code-coverage-value~1]
+    void testExtractPathCoverage(final int missed, final int covered, final float expectedPercentage) throws Exception {
+        final Path baseTestDir = BASE_TEST_DIR.resolve("project-with-coverage");
+        final Path targetDir = baseTestDir.resolve("target");
+        final Path siteDir = prepareSiteDir(targetDir);
+        createMetricsFile(siteDir, missed, covered);
+        // [itest -> qs~allowed-execution-time~1]
+        runMojoWithTimeout(baseTestDir, Duration.of(500, MILLIS));
+        assertThat(Files.readString(targetDir.resolve("metrics.json")), equalTo("""
+                {
+                    "coverage" : %.1f
+                }
+                """.formatted(expectedPercentage)));
+        cleanUpSiteDir(siteDir);
+    }
+
+    private static Path prepareSiteDir(Path targetDir) {
+        try {
+            final Path siteDir = targetDir.resolve("site");
+            deleteDirectoryRecursively(siteDir);
+            Files.createDirectories(siteDir);
+            return siteDir;
+        } catch (IOException exception) {
+            throw new IllegalStateException("Failed to prepare a clean site directory: " + exception.getMessage(),
+                    exception);
+        }
+    }
+
+    private static void deleteDirectoryRecursively(Path dir) throws IOException {
+        if (Files.exists(dir)) {
+            Files.walk(dir).map(Path::toFile).sorted(Comparator.reverseOrder()).forEach(File::delete);
+        }
+    }
+
+    private static void createMetricsFile(final Path siteDir, final int missed, final int covered) throws IOException {
+        Files.write(siteDir.resolve("jacoco.xml"), """
+                <report name="Example project">
+                    <counter type="BRANCH" missed="%d" covered="%d"/>
+                </report>
+                """.formatted(missed, covered).getBytes());
+    }
+
+    private static void runMojoWithTimeout(final Path projectDir, final Duration timeout) throws Exception {
```

Comment:
Nitpicking: After reading "timeout" I had expected that this would stopped the mojo after the timeout, but instead it just checks how long it took.